### ISSUE TITLE
Markers and shapes misbehaving

### ISSUE THREAD

**Issue by nikola-kocic** [zhenyanghua/MeasureTool-GoogleMaps-V3#22]
After instantiating MeasureTool, markers and shapes misbehave.
Examples:

- Rectangle cannot be moved by drag&drop.
- Markers placed outside initially visible part of the map cannot be moved by drag&drop and `mouseover` and `mouseout` events are not fired.

In attachment is sample code that reproduces the issue.
1. Load `index.html` in browser.
2. Move mouse over map and press key `s` to add rectangle.
3. Try moving the rectangle by drag&drop (this should work, but it's not).
4. Move mouse over map and press key `t` to add marker.
5. Try moving the marker by drag&drop (this works).
6. Pan the map to the north (across the river).
7. Once the map is panned across river, press key `t` to add marker.
8. Try moving the marker by drag&drop (this should work, but it's not).
9. Open `main.js` file in editor and change first line to `const USE_MEASURE_TOOL = false;`
10. Repeat steps 1 through 8. In this case everything works fine.

Tested in Firefox 57 and Chrome 62.

[measuretool-drag-and-drop-bug.zip](https://github.com/zhenyanghua/MeasureTool-GoogleMaps-V3/files/1482911/measuretool-drag-and-drop-bug.zip)

**Comment by KevinSJ** [comment]
I have this exact same issue.  Seems to me the cause is that the tool created an SVG overlay when binding to the  context menu which caused the mousedown/mouseover/mouseout event to be received by the overlay rather than the marker.

**Comment by KevinSJ** [comment]
I think I have fixed my issue of dragging by changing line 184 in index.js from 
```javascript 
this._svgOverlay = select(this._overlay.getPanes().overlayMouseTarget)
```
to 
```javascript 
this._svgOverlay = select(this._overlay.getPanes().overlayLayer)
```

**Comment by KevinSJ** [comment]
Seems I speak too soon, this will break the ability to remove the node from map.

**Comment by zhenyanghua** [comment]
#26 seems to point to the same cause.

**Comment by zhenyanghua** [comment]
This is caused by the svg pane overlay on top of google elements pane. I can think of a solution is to clear the overlay when a measure is ended, and reset the overlay when a measure is started. This should solve the problem.

**Comment by zhenyanghua** [comment]
Fixed in commit [5dacda](https://github.com/zhenyanghua/MeasureTool-GoogleMaps-V3/commit/5dacdaf323d35434ced01584fecb759b506865bd). Released in v0.0.14.
